### ISSUE TITLE
Make user preferences use the standard config service endpoints

### ISSUE THREAD

**Issue by dgrumieaux** [tsgrp/hpi#770]
Currently the user preferences module makes its save calls to the older RESTUserPreferencesService OpenContent controller. The user preferences module should be updated to use the same config endpoints as the rest of HPI.

The following additional items will either need to be coded or configured:

- making the /hpi/configs/UserPreferences path editable by any user (to support users adding/changing their config objects)
- making the config service properly call "save" vs "update" when a config already exists (the RESTUserPreferencesService handles this case whereas the RepositoryConfigService does not)

@benallenallen

**Comment by dgrumieaux** [comment]
Note - once this enhancement is complete, delete the RESTUserPreferencesService in OC.

**Comment by dgrumieaux** [comment]
Note that the SOLR HPI build will require this to be fixed in order to use User Preferences (or any other implementation of OpenContent using the FilesystemConfigService). Changing this to Upon Client Request.

@gsteimer

**Comment by gsteimer** [comment]
I'm moving this back to a 2.4 item.  Our enhancements for user preferences broke previously working functionality for HPI on Solr.

**Comment by gsteimer** [comment]
Another thought - in most HPI/Solr implementations, there's no concept of a "user".  Because we used a cookie before, we didn't have to worry about it.  But now that we've moved it to user preferences, I'm thinking we're going to run into more problems than just what endpoint to use.

At a minimum, we need to make sure that Solr implementations have access to:

- [ ] Saved Searches - added for HPI/Solr in late 2014
- [ ] Search Column preferences (order / visibility) - broken by change to user preferences
- [ ] Recent Objects - still a cookie

Recent objects is cookie-based now, so that should just work.  For the column preferences - would it be easier to somehow come up with a way to tie a "user" to a user preferences config somehow?  Or would it be easier to have the HPI code revert to using a cookie if user preferences aren't available in the back-end repo?

@dgrumieaux - let me know if this should really be two issues.  One for the original request to use one endpoint, and one to get HPI/Solr features working again.

**Comment by dgrumieaux** [comment]
@gsteimer - this makes sense as two different issues. This issue here is more of an OC/repository issue (the only HPI change is to switch a URL). The new issue, as I see it, would be to enable a setting in HPI to save userpreferences as a cookie instead of using the repository.

**Comment by gsteimer** [comment]
New issue created for HPI/Solr and user preferences: #771

**Comment by gsteimer** [comment]
Note from @dgrumieaux - when implementing this, one difference between normal configs and user preferences is that we minor version normal configs upon save, however we don't want to version user preferences

**Comment by gsteimer** [comment]
@dgrumieaux - from discussions during the weekly meeting, we're thinking that we don't want user prefs to use the standard endpoints.  Basically - we need to somehow ignore 403 errors (see #989) for user prefs, but we don't want to do that for normal config calls.

Additional thoughts - @mikeblum @benallenallen @nwoodbine?

**Comment by dgrumieaux** [comment]
Is the other option adding a silent flag into the standard config service endpoint? I don't think I like that idea, but I almost like it more than having duplicated code.

**Comment by gsteimer** [comment]
@dgrumieaux - I think the other concern was that we may want to start using user prefs for other tools besides HPI (ex: OA).  So tying user prefs to HPI's config service may not be ideal.  Is that correct @m-brouillette?
